src/transformers/trainer.py

```diff
@@ -484,11 +489,12 @@ def __init__(
         ):
             self.place_model_on_device = False
 
-        default_collator = default_data_collator if tokenizer is None else DataCollatorWithPadding(tokenizer)
+        default_collator = DataCollatorWithPadding(tokenizer) if tokenizer is not None else default_data_collator
```

Comment:
might not be BC if people pass a ImageProcessor. Let's deprecate passing image processors checking the type

Reply:
If people currently pass an image processor as tokenizer, then they need to pass a data collator as otherwise they'll hit the issue mentioned at https://github.com/huggingface/transformers/issues/29790.